openmetadata-ui/src/main/resources/ui/playwright/utils/lineage.ts

```diff
@@ -0,0 +1,386 @@
+    '[data-testid="suggestion-node"] input'
+  );
+  await suggestionNodeInput.dispatchEvent('click');
+  await suggestionNodeInput.fill(toNodeName);
+  await page
+    .locator(`[data-testid="node-suggestion-${toNodeFqn}"]`)
+    .dispatchEvent('click');
+
+  await dragConnection(
+    page,
+    `lineage-node-${fromNodeFqn}`,
+    `lineage-node-${toNodeFqn}`
+  );
+};
+
+export const verifyNodePresent = (page: Page, node: EntityClass) => {
+  const nodeFqn = get(node, 'entityResponseData.fullyQualifiedName');
+  const name = get(node, 'entityResponseData.name');
+  const lineageNode = page.locator(`[data-testid="lineage-node-${nodeFqn}"]`);
+
+  expect(lineageNode).toBeVisible();
+
+  const entityHeaderName = lineageNode.locator(
+    '[data-testid="entity-header-name"]'
+  );
+
+  expect(entityHeaderName).toHaveText(name);
+};
+
+export const setupEntitiesForLineage = async (
+  page: Page,
+  currentEntity:
+    | TableClass
+    | DashboardClass
+    | TopicClass
+    | MlModelClass
+    | ContainerClass
+    | SearchIndexClass
+) => {
+  const entities = [
+    new TableClass(),
+    new DashboardClass(),
+    new TopicClass(),
+    new MlModelClass(),
+    new ContainerClass(),
+    new SearchIndexClass(),
+  ] as const;
+
+  const { apiContext, afterAction } = await getApiContext(page);
+  for (const entity of entities) {
+    await entity.create(apiContext);
+  }
+
+  await currentEntity.create(apiContext);
+
+  const cleanup = async () => {
+    await currentEntity.delete(apiContext);
+    for (const entity of entities) {
+      await entity.delete(apiContext);
+    }
+    await afterAction();
+  };
+
+  return { currentEntity, entities, cleanup };
+};
+
+export const editPipelineEdgeDescription = async (
+  page: Page,
+  fromNode: EntityClass,
+  toNode: EntityClass,
+  pipelineData,
+  description: string
+) => {
+  const fromNodeFqn = get(fromNode, 'entityResponseData.fullyQualifiedName');
+  const toNodeFqn = get(toNode, 'entityResponseData.fullyQualifiedName');
+
+  await page.click(
+    `[data-testid="pipeline-label-${fromNodeFqn}-${toNodeFqn}"]`
+  );
+  await page.locator('.edge-info-drawer').isVisible();
+  await page
+    .locator('.edge-info-drawer [data-testid="Edge"] a')
+    .filter({ hasText: pipelineData.name });
+
+  await page.click('.edge-info-drawer [data-testid="edit-description"]');
+  await page.locator('.ProseMirror').first().click();
+  await page.locator('.ProseMirror').first().clear();
+  await page.locator('.ProseMirror').first().fill(description);
+  const descRes = page.waitForResponse('/api/v1/lineage');
+  await page.getByTestId('save').click();
+  await descRes;
+
+  await expect(
+    page.getByTestId('asset-description-container').getByRole('paragraph')
+  ).toContainText(description);
+};
+
+const verifyPipelineDataInDrawer = async (
+  page: Page,
+  fromNode: EntityClass,
+  toNode: EntityClass,
+  pipelineItem: PipelineClass,
+  bVerifyPipelineLineage: boolean
```

Comment:
the purpose for variable `bVerifyPipelineLineage` is not very clear from the name